### DISCUSSION TITLE
javascript includes

### DISCUSSION THREAD

**Discussion by Lveronneau** [ant-media/Ant-Media-Server#6776] [FAQs]
good evening everyone !

developping a videochat webapp and trying to minimise the use of includes specially jquery !

I am using import {WebRTCAdaptor} from "https://cdn.skypack.dev/@antmedia/webrtc_adaptor@SNAPSHOT";  top of my module section.

question 1 - do I need those includes ?
      <script src="//ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>        
      <script src="https://video2.meetstream.com:5443/LiveApp/js/external/adapter-latest.js"></script>
      <script src="https://video2.meetstream.com:5443/LiveApp/js/external/notify.min.js" crossorigin="anonymous"></script>

because when i dont use them, the code still works but  i get the error :
==================================================================
"Video sender not found to change bandwidth. Streaming may not be active	
remotePeerConnection.<computed>.oniceconnectionstatechange	@	webrtc_adaptor.js:2073"
==================================================================

quesiton 2 - can i replace the import of : https://cdn.skypack.dev/@antmedia/webrtc_adaptor@SNAPSHOT"
by a file i can host on my own server ? like : import { WebRTCAdaptor } from "./js/webrtc_adaptor.js";  ??

thanks

**Comment by lastpeony** [comment]
Get rid of script tags and just use
`npm i @antmedia/webrtc_adaptor
`
`import { WebRTCAdaptor } from '@antmedia/webrtc_adaptor';
`
https://www.npmjs.com/package/@antmedia/webrtc_adaptor

Im not a fan of script cdn style imports either.

**Comment by Lveronneau** [comment]
I still get this error : 
![Screenshot from 2024-11-08 10-52-33](https://github.com/user-attachments/assets/33a84bac-be58-4d54-ad8d-f62e9284cc92)

any reason why ?

thanks

**Reply by rahul78275** [reply]
Hello @Lveronneau 



>  It tries to change the bandwidth for the playing stream and there is no video sender for the incoming stream.
> The functionality(change bandwidth) works for the outgoing stream.

You can ignore this warning error. (Developer don't care notices and warnings 😆 ) Here is the [reference](https://github.com/ant-media/Ant-Media-Server/issues/2698)

**Comment by Anatolii-Ovcharuk** [comment]
Good evening! Here’s some help with your questions:

    External Script Requirements: You may not need all these scripts, depending on the WebRTCAdaptor's requirements and your specific implementation:
        jQuery: If you're not directly using jQuery in your code or WebRTCAdaptor doesn’t depend on it, you can skip this include.
        Adapter-latest.js: This is usually for WebRTC support in older browsers, so if you're targeting modern browsers, you might not need it. However, WebRTC libraries sometimes rely on it to handle cross-browser WebRTC differences.
        Notify.js: This script enables notifications, so if you don’t specifically call notify() functions, you should be able to remove it.

    The error about "Video sender not found to change bandwidth" might be an unrelated issue, possibly tied to how the WebRTCAdaptor manages connection state changes, especially if a connection fails or drops. Verify if the connection to your video stream has any issues with initialization or bandwidth.

    Hosting WebRTCAdaptor Locally: Yes, you can host a local copy of webrtc_adaptor.js by downloading the file and referencing it as import { WebRTCAdaptor } from "./js/webrtc_adaptor.js";. This gives you full control over its version and eliminates external dependency, but remember that it may not update automatically if the library is updated on Skypack.
